src/enclave/enclave.h

```diff
@@ -256,6 +256,22 @@ namespace enclave
             idling_start_time = time_now;
           }
 
+          // If we have pending thread messages, handle them now (and don't
```

Comment:
I'm not too familiar with this code: why did this get moved from `messaging.h`?

Reply:
This removes the dependency of `messaging.h` on `thread_messaging.h`, so we can test the "read messages from the ringbuffer" logic without any thread state initialisation.

This _does_ change the semantics slightly. Thread 0 inside the enclave may now do more reading-from-the-ringbuffer jobs before it handles any thread tasks. But the difference is minimal; its now "read everything from the ringbuffer, try again and see its empty, try a thread task" rather than "read everything from the ringbuffer, try a thread task".